apisix/plugins/authz-keycloak.lua

```diff
@@ -52,8 +55,7 @@ local schema = {
         keepalive_timeout = {type = "integer", minimum = 1000, default = 60000},
         keepalive_pool = {type = "integer", minimum = 1, default = 5},
         ssl_verify = {type = "boolean", default = true},
-    },
-    required = {"token_endpoint"}
```

Comment:
We can do it in jsonschema: https://github.com/apache/apisix/blob/3e19b06293463f52c8d26f0958542fc475ff0fcd/apisix/plugins/echo.lua#L48-L51
And need to a test for it.

Reply:
Right, I'm not yet a JSON schema pro.

Reply:
Fixed.

Reply:
@jenskeiner 
Sorry for my mistake! It should be `oneOf` instead of `anyOf`. 😢

Reply:
Right, true. Will change.

Reply:
Hmm, I think we should actually use `anyOf` since the explicit endpoint URL will be preferred if both options are given. The full schema unit tests already uses both options so would have to touch that again as well if we make this XOR.

I have future dev work in mind that would require using an additional endpoint. In that case, you may want to use discovery for one endpoint and an explicit value for the other. But, admittedly, that's a rather strange case. In any sane environment, you would always use only discovery or only explicit endpoints, I think.

Let me know your thoughts @spacewander.

Reply:
OK